src/swap/saga.ts

```diff
@@ -124,30 +140,47 @@ export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
       TAG,
       `Approving ${amountToApprove} of ${sellTokenAddress} for address: ${allowanceTarget}`
     )
+    const beforeApproveTimestamp = Date.now()
     yield call(
       sendApproveTx,
       sellTokenAddress,
       amountToApprove,
       allowanceTarget,
       swapApproveContext
     )
+    const afterApproveTimestamp = Date.now()
+    sendApprovalElapsedTimeInMs = afterApproveTimestamp - beforeApproveTimestamp
```

Comment:
why are we interested in this? i thought the interesting bit would be the time between when the quote response was received and beforeApproveTimestamp?

Reply:
I wasn't sure... I thought that adding these won't harm, and maybe they are useful to see some issues.
Like if the `quoteToTransactionElapsedTimeInMs`, it would be great to understand where the bottleneck was.

I'm fine if you want me to remove them. We don't really care how much it took until we send the approval, since that should work always.